sphinx_gallery/tests/test_binder.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Author: Chris Holdgraf
+# License: 3-clause BSD
+"""
+Testing the binder badge functionality
+"""
+from __future__ import division, absolute_import, print_function
+import os
+import tempfile
+import sys
+from copy import deepcopy
+
+import pytest
+
+from sphinx_gallery.binder import (gen_binder_rst, gen_binder_url,
+                                   check_binder_conf)
+from sphinx_gallery.utils import _TempDir
+
+
+def test_binder():
+    """Testing binder URL generation and checks."""
+    file_path = 'myfile.py'
+    conf1 = {'url': 'http://test1.com', 'org': 'org',
+             'repo': 'repo', 'branch': 'branch',
+             'dependencies': '../requirements.txt'}
+    url = gen_binder_url(file_path, conf1)
+    assert url == 'http://test1.com/v2/gh/org/repo/branch?filepath=_downloads/myfile.ipynb'
+
+    # URL must have http
+    with pytest.raises(ValueError):

```

Comment:
I am a big fan of testing the error message as well as the exception type. You can use something like (taken from test_notebook.py):

```py
def test_missing_file():
    with pytest.raises(FileNotFoundError) as excinfo:
        python_to_jupyter_cli(['nofile.py'])
    excinfo.match(r'No such file or directory.+nofile\.py')
```

Same comment applies to a few `pytest.raises` below.

Reply:
that's a good idea, hadn't thought of that